core/solver/multigrid.cpp

```diff
@@ -875,16 +875,12 @@ Multigrid::Multigrid(const Multigrid::Factory* factory,
           stop::combine(factory->get_parameters().criteria)},
       parameters_{factory->get_parameters()}
 {
+    this->validate();
```

Comment:
I guess `validate` doesn't check the level or solver selector?

Reply:
No, it only checks the length of mg_level and smoother, and no nullptr in the mg_level list